### PR TITLE
Added rule for safe cast instead of if-else-null

### PR DESCRIPTION
**Pull request by schalkms**
#197 feature

### CODE REVIEW

**Comment by arturbosch** [comment]
Looks good! Will check it out tomorror. Thank you very much!

**Comment by vanniktech** [comment]
Maybe also add a test case for a negative case?

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/providers/StyleGuideProvider.kt [hunk 1]
```diff
@@ -3,11 +3,7 @@ package io.gitlab.arturbosch.detekt.rules.providers
 import io.gitlab.arturbosch.detekt.api.Config
 import io.gitlab.arturbosch.detekt.api.RuleSet
 import io.gitlab.arturbosch.detekt.api.RuleSetProvider
-import io.gitlab.arturbosch.detekt.rules.style.ForbiddenComment
-import io.gitlab.arturbosch.detekt.rules.style.MaxLineLength
-import io.gitlab.arturbosch.detekt.rules.style.NamingConventionViolation
-import io.gitlab.arturbosch.detekt.rules.style.NewLineAtEndOfFile
-import io.gitlab.arturbosch.detekt.rules.style.WildcardImport
+import io.gitlab.arturbosch.detekt.rules.style.*
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/providers/StyleGuideProvider.kt]: Pls import the code-style.xml when working on detekt. The codebase should not have wildcards.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/SafeCastTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import io.gitlab.arturbosch.detekt.api.Finding
+import io.gitlab.arturbosch.detekt.test.lint
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+
+class SafeCastTest {
+
+	@Test
+	fun findNegated() {
+		val code = "fun test() {\n" +
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/SafeCastTest.kt]: Use multiline strings in your test cases. It's better to read :)

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/SafeCastTest.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		Assertions.assertThat(findings.size).isEqualTo(1)
+	}
+
+	private fun findNone(code: String) {
+		val findings = retrieveFindings(code)
+		Assertions.assertThat(findings.size).isEqualTo(0)
+	}
+
+	private fun retrieveFindings(code: String): List<Finding> {
+		return SafeCast().lint(code)
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/SafeCastTest.kt]: Extend `RuleTest` and override the field `rule` with SafeCast. `rule.lint(code)` can then be inlined
